### ISSUE TITLE
Feature: Possibility to forget/prune all snapshots at once

### ISSUE THREAD

**Issue by martinseener** [restic/restic#1610]
## Output of `restic version`
Version indipended but tested with latest stable and beta.

## How did you run restic exactly?
does not matter

## What backend/server/service did you use to store the repository?
SFTP

## Expected behavior
restic -p passfile -r sftp:user@server.com forget --prune all

## Actual behavior
`all` does not exist

## Steps to reproduce the behavior
try it.

## Do you have any idea what may have caused this?
not implemented yet - this is a feature request

## Do you have an idea how to solve the issue?
Sometimes it's necessary to purge a whole repo (without purging the repo but all snapshots within). For this a parameter `all` for `forget` would be nice.

## Did restic help you or made you happy in any way?
I'm totally happy with restic!

**Comment by rawtaz** [comment]
Please clarify; Are you asking for a feature that can `forget` *all* snapshots in your repository? It seems so, by your example command.

In that case, this would effectively remove all the backups in your repository, and in that case, why don't you just delete the entire repository instead?

**Comment by martinseener** [comment]
Exactly, i want to purge all snapshots. Sure, i could erase that repo manually and re-init it but that's not as cool. Another way would be to run "latest" for forget until nothing more there is to forget ;)

**Comment by rawtaz** [comment]
@martinseener You are confusing things here, `forget` is one thing, `prune` is another. The former removes snapshots from the list of snapshots in the repository. The latter just cleans up the data that these snapshots did occupy. There's no `purge` command, let's stick to the right keywords :)

So "pruning all snapshots", if that's what you mean, is what the `prune` command does by default. And "forgetting all snapshots", which I asked if that's what you meant, is something that would effectively remove all your backups, at which point I ask why you don't instead just delete the entire repository?

**Comment by martinseener** [comment]
Sorry, @rawtaz . You're right. So i mean `forget` all snapshots, yes. I never use forget without prune, so that was essentially the "same" for me, but yes, something like `restic forget --prune all` which forgets all snapshots (and prunes them) would be nice and time-saving.

**Comment by rawtaz** [comment]
Right :) But still, the question remains. If you want to effectively delete all snapshots in your repository, and also all the data that they occupied in the repository, then a much quicker way is to just delete your entire repository. Why don't you want to do that instead? I fail to see the point of the suggested feature, that's all.

**Comment by martinseener** [comment]
Because i then have to manually login and delete the repo files/folders and then do another `init` which is annoying. another way could be to have something like `restic init --force` to force recreating the repo ;)

**Comment by rawtaz** [comment]
I see. I think your second suggestion might be more to the point, reinitializing an existing repo. Let's hear what others think!

**Comment by martinseener** [comment]
Agree, i also think this might save some bandwith and cpu time, because it's just building a new repo instead of doing a time-consuming forget. Also `--force` may be a bit misleading if someone fails to init a repo and then want to force it. Maybe something like `restic init --overwrite` would make sense.

**Comment by pvgoran** [comment]
Cleaning a repository (instead of re-initializing it) could be useful if one wants to preserve the repository password.

**Comment by martinseener** [comment]
@pvgoran but the repository password isn't saved in the repository itself. This doesn't make any sense to me!?

**Comment by pvgoran** [comment]
@martinseener The password is not saved in a repository, but a repository is tied to its password. If a repository is recreated, the password needs to be retyped, which can lead to unintentional password change. If a repository is preserved, it's guaranteed to have the same password as before.

**Comment by martinseener** [comment]
@pvgoran That's true, but only if you type it. I'm using a password file (with `-p`) and that's mostly the case for automated backups, or do you ever do it manually and type in your password? I agree on that possible problem, so maybe we should consider both solutions as valid.

**Comment by silverwind** [comment]
I feel like is should be a single command to forget all snapshots. Currently, it needs something like this:

```bash
restic forget --keep-last 1 --prune
restic snapshots # copy last id
restic forget <last id> --prune
```

While it could be simply something like

```bash
restic forget --all --prune
```

**Comment by rawtaz** [comment]
@silverwind Thanks for your input, but this has been discussed and is settled upon. It's deemed more important to make sure people don't accidentally shoot themselves in the foot, than having them type two lines less than they otherwise would have, the very few times they want to completely empty a repository. It's arguably very rare that you need to do this. It will not change at this point, sorry if this is not to your liking.

**Comment by fd0** [comment]
For the record, you can use `restic forget latest` instead of having to supply the snapshot ID.
